server/db/mongodb/adapter.go

```diff
@@ -1719,22 +1723,24 @@ func (a *adapter) getFindPipeline(req, opt []string) (map[string]struct{}, b.A)
 		b.M{"$sort": b.M{"matchedTagsCount": -1}},
 	}
 
-	if len(req) > 0 {
-		var reqTags []interface{}
-		for _, tag := range req {
-			reqTags = append(reqTags, tag)
-		}
+	if len(allReq) > 0 {
+		for _, l := range req {
+			var reqTags []interface{}
+			for _, tag := range l {
+				reqTags = append(reqTags, tag)
+			}
 
-		// Filter out documents where 'tags' intersection with 'reqTags' is an empty array
-		pipeline = append(pipeline,
-			b.M{"$match": b.M{"$expr": b.M{"$ne": b.A{b.M{"$size": b.M{"$setIntersection": b.A{"$tags", reqTags}}}, 0}}}})
+			// Filter out documents where 'tags' intersection with 'reqTags' is an empty array
+			pipeline = append(pipeline,
+				b.M{"$match": b.M{"$expr": b.M{"$ne": b.A{b.M{"$size": b.M{"$setIntersection": b.A{"$tags", reqTags}}}, 0}}}})
```

Comment:
I'm not sure this is correct. The idea is to have a query like `(term OR rewritten_term) AND (term2 OR rewritten_term2)...` when it's just `term AND rewritten_term AND term2 AND rewritten_term2...`

Reply:
It intersects rows tags with `[term, rewrittenTerm]` and checks if the intersection is empty. Which is equivalent to matching terms to `term OR rewrittenTerm`.
Am I wrong?

Reply:
Yes, you are right, thanks. Resolved.